module/VuFind/src/VuFind/ILS/Driver/AbstractAPI.php

```diff
@@ -133,17 +153,21 @@ public function makeRequest(
                 $client->setParameterPost($params);
             }
         }
-        $response = $client->send();
-        switch ($response->getStatusCode()) {
-        case 400:
-            throw new BadRequest($response->getBody());
-        case 401:
-        case 403:
-            throw new Forbidden($response->getBody());
-        case 404:
-            throw new RecordMissing($response->getBody());
-        case 500:
-            throw new ILSException("500: Internal Server Error");
+        try {
+            $response = $client->send();
+        } catch (\Exception $e) {
+            $this->logError("Unexpected " . get_class($e) . ": " . $e->getMessage());
```

Comment:
I'd log `(string)$e` instead of `$e->getMessage()`. I've found it to be very useful to get a stack leading to the exception.